### PR TITLE
ENH: Improve numerical precision of logistic function

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR improves the numerical precision of the logistic loss function by splitting the term into separate parts that are then multiplied and summed as needed.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

File: cpp/daal/src/algorithms/objective_function/logistic_loss/logistic_loss_dense_default_batch_impl.i [hunk 1]
```diff
@@ -431,10 +434,13 @@ services::Status LogLossKernel<algorithmFPType, method, cpu>::doCompute(const Nu
...
                 else
                 {
+                    algorithmFPType l1NormBeta = 0;
+                    PRAGMA_FORCE_SIMD
                     for (size_t i = 1; i < nBeta; ++i)
                     {
-                        value += (b[i] < 0 ? -b[i] : b[i]) * parameter->penaltyL1;
+                        l1NormBeta += b[i] < 0 ? -b[i] : b[i];
                     }
+                    value += parameter->parameter->penaltyL1 * l1NormBeta;
```
Comment (david-cortes-intel) [inline cpp/daal/src/algorithms/objective_function/logistic_loss/logistic_loss_dense_default_batch_impl.i]:
```suggestion
                    value += parameter->penaltyL1 * l1NormBeta;
```

File: cpp/daal/src/externals/service_blas_ref.h [hunk 2]
```diff
@@ -129,6 +129,8 @@ struct OpenBlas<double, cpu>
     }
 
     static void xscal(const DAAL_INT * n, const double * a, double * x, const DAAL_INT * incx) { dscal_(n, a, x, incx); }
+
+    static double xasum(const DAAL_INT * n, const double * x, const DAAL_INT * incx) { return dasum_(n, a, x, incx); }
```
Comment (Vika-F) [inline cpp/daal/src/externals/service_blas_ref.h]:
```suggestion
    static double xasum(const DAAL_INT * n, const double * x, const DAAL_INT * incx) { return dasum_(n, x, incx); }
```

File: cpp/daal/src/algorithms/objective_function/logistic_loss/logistic_loss_dense_default_batch_impl.i [hunk 3]
```diff
@@ -431,10 +434,10 @@ services::Status LogLossKernel<algorithmFPType, method, cpu>::doCompute(const Nu
                 }
                 else
                 {
-                    for (size_t i = 1; i < nBeta; ++i)
-                    {
-                        value += (b[i] < 0 ? -b[i] : b[i]) * parameter->penaltyL1;
-                    }
+                    DAAL_INT nBeta_minus_one         = nBeta - 1;
+                    DAAL_INT one                     = 1;
```
Comment (Vika-F) [inline cpp/daal/src/algorithms/objective_function/logistic_loss/logistic_loss_dense_default_batch_impl.i]:
```suggestion
                    const DAAL_INT nBeta_minus_one         = nBeta - 1;
                    const DAAL_INT one                     = 1;
```
